storage/scanner.go

```diff
@@ -71,6 +71,7 @@ type storeStats struct {
 // complete approximately one full scan per interval. Each range is
 // tested for inclusion in a sequence of prioritized range queues.
 type rangeScanner struct {
+	store    *Store         // The store this scanner is being called from
```

Comment:
Instead of passing the store, let's try to keep the abstraction cleaner by passing in a function. Maybe something like:

    scanFn func() // Function called at each complete scan iteration